### PR TITLE
add tests for #94502

### PR DESCRIPTION
**Pull request by lcnr**
cc #94552

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
@bors r-
looks like this failed in a rollup, not sure why it didn't fail in pre-merge ci though, hmm :thinking: 
https://github.com/rust-lang/rust/pull/94567#issuecomment-1058378405

File: src/test/ui/nll/lint-no-err.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// check-pass
```
Comment (matthiaskrgr) [inline src/test/ui/nll/lint-no-err.rs:1]: oh is it because of `check-pass`

Comment (Dylan-DPC) [reply src/test/ui/nll/lint-no-err.rs:1]: but i don't see why it wouldn't be picked by the CI then

Comment (lcnr) [reply src/test/ui/nll/lint-no-err.rs:1]: this borrowcheck warning is a hard error with `nll` enabled, we don't test nll in ci

**Pull request merged**